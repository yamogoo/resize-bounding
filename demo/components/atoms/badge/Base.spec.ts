import { shallowMount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";

import Base from "./Base.vue";

const DEFAULT_PROPS = {
  label: "Label",
  value: "Value",
};

describe("Base", () => {
  test(`should display "label" (${DEFAULT_PROPS.label}), "value" (${DEFAULT_PROPS.value})`, () => {
    const wrapper = shallowMount(Base, {
      props: {
        ...DEFAULT_PROPS,
      },
    });

    const badgeEl = wrapper.find(`[data-testid="ui-badge"]`);
    const labelText = badgeEl.find(`[data-testid="ui-badge-label"]`).text();
    const valueText = badgeEl.find(`[data-testid="ui-badge-value"]`).text();

    expect(labelText).toBe(DEFAULT_PROPS.label);
    expect(valueText).toBe(DEFAULT_PROPS.value);

    expect(labelText).toMatchSnapshot();
    expect(valueText).toMatchSnapshot();
  });
});
