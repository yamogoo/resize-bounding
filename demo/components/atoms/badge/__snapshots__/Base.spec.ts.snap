// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base > should display "label" (Label), "value" (Value) 1`] = `"Label"`;

exports[`Base > should display "label" (Label), "value" (Value) 2`] = `"Value"`;
